core/codenvy-hosted-machine-token-service/src/main/java/com/codenvy/service/machine/token/interceptor/MachineTokenInterceptor.java

```diff
@@ -0,0 +1,54 @@
+/*
+ *  [2012] - [2016] Codenvy, S.A.
+ *  All Rights Reserved.
+ *
+ * NOTICE:  All information contained herein is, and remains
+ * the property of Codenvy S.A. and its suppliers,
+ * if any.  The intellectual and technical concepts contained
+ * herein are proprietary to Codenvy S.A.
+ * and its suppliers and may be covered by U.S. and Foreign Patents,
+ * patents in process, and are protected by trade secret or copyright law.
+ * Dissemination of this information or reproduction of this material
+ * is strictly forbidden unless prior written permission is obtained
+ * from Codenvy S.A..
+ */
+package com.codenvy.service.machine.token.interceptor;
+
+import com.codenvy.service.machine.token.MachineTokenRegistry;
+
+import org.aopalliance.intercept.MethodInterceptor;
+import org.aopalliance.intercept.MethodInvocation;
+import org.eclipse.che.api.core.model.workspace.Workspace;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+/**
+ * Intercepts calls to workspace start/stop methods and
+ * creates machine authorization token in the registry.
+ *
+ * @author Max Shaposhnik (mshaposhnik@codenvy.com)
+ */
+@Singleton
+public class MachineTokenInterceptor implements MethodInterceptor {
```

Comment:
Please add tests which check compatibility of this interceptor [example](https://github.com/codenvy/codenvy/blob/79917d88fa4a04ca4f7ab4795e610249ed3fbebe/core/codenvy-hosted-platform-api-impl/src/test/java/com/codenvy/api/account/AddDefaultAccountIdInterceptorTest.java), because it is impossible to figure out whether everything is okay.